docs/reference/elasticsearch/rest-apis/update-document.md

```diff
@@ -0,0 +1,264 @@
+    }
+  }
+}
+```
+
+You could also remove a tag from the list of tags. The Painless
+function to `remove` a tag takes the array index of the element
+you want to remove. To avoid a possible runtime error, you first need to
+make sure the tag exists. If the list contains duplicates of the tag, this
+script just removes one occurrence.
+
+```console
+POST test/_update/1
+{
+  "script": {
+    "source": "if (ctx._source.tags.contains(params.tag)) { ctx._source.tags.remove(ctx._source.tags.indexOf(params.tag)) }",
+    "lang": "painless",
+    "params": {
+      "tag": "blue"
+    }
+  }
+}
+```
+
+You can also add and remove fields from a document. For example, this script
+adds the field `new_field`:
+
+```console
+POST test/_update/1
+{
+  "script" : "ctx._source.new_field = 'value_of_new_field'"
+}
+```
+
+Conversely, this script removes the field `new_field`:
+
+```console
+POST test/_update/1
+{
+  "script" : "ctx._source.remove('new_field')"
+}
+```
+% TEST[continued]
+
+The following script removes a subfield from an object field:
+
+```console
+PUT test/_doc/1?refresh
+{
+  "my-object": {
+    "my-subfield": true
+  }
+}
+```
+
+```console
+POST test/_update/1
+{
+  "script": "ctx._source['my-object'].remove('my-subfield')"
+}
+```
+% TEST[continued]
+
+Instead of updating the document, you can also change the operation that is
+executed from within the script. For example, this request deletes the doc if
+the `tags` field contains `green`, otherwise it does nothing (`noop`):
+
+```console
+POST test/_update/1
+{
+  "script": {
+    "source": "if (ctx._source.tags.contains(params.tag)) { ctx.op = 'delete' } else { ctx.op = 'noop' }",
+    "lang": "painless",
+    "params": {
+      "tag": "green"
+    }
+  }
+}
+```
+
+## Update part of a document [update-part-document]
+
+The following partial update adds a new field to the
+existing document:
+
+```console
+POST test/_update/1
+{
+  "doc": {
+    "name": "new_name"
+  }
+}
+```
+
+If both `doc` and `script` are specified, then `doc` is ignored. If you
+specify a scripted update, include the fields you want to update in the script.
+
+
+## Detect noop updates [detect-noop-updates]
+
+By default updates that don't change anything detect that they don't change
+anything and return `"result": "noop"`:
+
+```console
+POST test/_update/1
+{
+  "doc": {
+    "name": "new_name"
+  }
+}
+```
+% TEST[continued]
+
+If the value of `name` is already `new_name`, the update
+request is ignored and the `result` element in the response returns `noop`:
+
+```console
+
+{
+   "_shards": {
+        "total": 0,
+        "successful": 0,
+        "failed": 0
+   },
+   "_index": "test",
+   "_id": "1",
+   "_version": 2,
+   "_primary_term": 1,
+   "_seq_no": 1,
+   "result": "noop"
+}
+```
+
+You can disable this behavior by setting `"detect_noop": false`:
+
+```console
+POST test/_update/1
+{
+  "doc": {
+    "name": "new_name"
+  },
+  "detect_noop": false
+}
+```
+
+## Upsert [upsert]
```

Comment:
Can improve heading here to explain what user value is and use verb phrase for parallel structure